addons/addon-base-raas/packages/base-raas-services/lib/environment/service-catalog/environment-sc-service.js

```diff
@@ -390,6 +395,14 @@ class EnvironmentScService extends Service {
     // Validate input
     await validationService.ensureValid(environment, createSchema);
 
+    // If the AppStream feature is enabled, verify that update request doesn't include a cidr
+    if (this.isAppStreamEnabled()) {
+      if (environment.cidr) {
+        throw this.boom.badRequest('Cannot specify CIDR when AppStream is enabled', true);
+      }
+      delete environment.cidr;
```

Comment:
If the code made it to this line, then there is no `cidr` field in the object, right?

Reply:
`cidr` key can be present but be undefined. From API point of view we are just saying both `cidr` not being there or `cidr` being there but being undefined are acceptable.